docs/rules/0123/resource-definition-type-name.md

```diff
@@ -0,0 +1,68 @@
+---
+rule:
+  aip: 123
+  name: [core, '0123', resource-definition-type-name]
+  summary: Resource type names must be of the form {Service Name}/{Type}.
+permalink: /123/resource-definition-type-name
+redirect_from:
+  - /0123/resource-definition-type-name
+---
+
+# Resource type name
+
+This rule enforces that files that define a resource with the
+`google.api.resource_definition` annotation have a properly formatted `type`, as
```

Comment:
Same re, not mentioned in the AIP.

Reply:
Opened https://github.com/aip-dev/google.aip.dev/issues/865